### ISSUE TITLE
no definition was found for superclass “RangedData”

### ISSUE THREAD

**Issue by oriolarques** [benilton/oligoClasses#1]
Hi,
I have been using the oligo package for many years but today, with the release of R 4.0.0, I encountered the following problem when trying to reinstall it:

**Error in reconcilePropertiesAndPrototype(name, slots, prototype, superClasses,  : 
  no definition was found for superclass “RangedData” in the specification of class “RangedDataCopyNumber”**

I would really apreciate any feedbak because I cannot find a way through it.

Thanks.

I enclose the code for the whole installation process:

Bioconductor version 3.12 (BiocManager 1.30.10), R 4.0.0 (2020-04-24)
Installing package(s) 'oligo'
also installing the dependency ‘oligoClasses’
Package which is only available in source form, and may need compilation of C/C++/Fortran: ‘oligo’

installing the source packages ‘oligoClasses’, ‘oligo’

trying URL 'https://bioconductor.org/packages/3.12/bioc/src/contrib/oligoClasses_1.51.0.tar.gz'
Content type 'application/x-gzip' length 373251 bytes (364 KB)

downloaded 364 KB

trying URL 'https://bioconductor.org/packages/3.12/bioc/src/contrib/oligo_1.53.0.tar.gz'
Content type 'application/x-gzip' length 28421038 bytes (27.1 MB)

downloaded 27.1 MB

* installing *source* package ‘oligoClasses’ ...
** using staged installation
** R
** data
** inst
** byte-compile and prepare package for lazy loading
**Error in reconcilePropertiesAndPrototype(name, slots, prototype, superClasses,  : 
  no definition was found for superclass “RangedData” in the specification of class “RangedDataCopyNumber”**
Error: unable to load R code in package ‘oligoClasses’
Execution halted
ERROR: lazy loading failed for package ‘oligoClasses’
* removing ‘/Library/Frameworks/R.framework/Versions/4.0/Resources/library/oligoClasses’
ERROR: dependency ‘oligoClasses’ is not available for package ‘oligo’
* removing ‘/Library/Frameworks/R.framework/Versions/4.0/Resources/library/oligo’

The downloaded source packages are in
	‘/private/var/folders/tt/f8_2bryn5vbfknlppg9xzsc40000gn/T/Rtmptk1LE5/downloaded_packages’
Warning messages:
1: In install.packages(...) :
  installation of package ‘oligoClasses’ had non-zero exit status
2: In install.packages(...) :
  installation of package ‘oligo’ had non-zero exit status

**Comment by oriolarques** [comment]
I managed to install it by downloading the source package and install it with the terminal using $R CMD INSTALL oligoClasses_1.50.0.tar. However, the error keeps appearing.

**Comment by jmacdon** [comment]
The `RangedData` class has been made defunct in `IRanges`. All the functions in `oligoClasses` that relied on that class were already defunct and I just made them deprecated in the devel version on Bioconductor. I used to have commit privileges to Benilton's github, but no longer seem to, so cannot (yet) push those changes here.

But in about 24 hours the devel version should get rebuilt on Bioconductor, and you will be able to get it using `BiocManager::install`.
